src/renderer/components/layout/top-bar/top-bar.tsx

```diff
@@ -84,19 +81,15 @@ const NonInjectedTopBar = (({ items, children, ...rest }: Props & Dependencies)
   };
 
   const minimizeWindow = () => {
-    window.minimize();
+    requestMain(IpcMainWindowEvents.WINDOW_ACTION, "minimize");
```

Comment:
Since `requestMain(channel: string, ...args: any[])` doesn't do type-checking for view type (e.g. "minimize", "window", etc.) + in order to reduce calling `requestMain` directly we could wrap all these things into "actions" with fully type checking for arguments, e.g.:

```ts
// common/ipc/window.actions.ts

export function windowAction(type: 'backgroundPage' | 'window' | 'browserView' | 'remote' | 'webview' | 'offscreen' = "window") {
  requestMain(IpcMainWindowEvents.WINDOW_ACTION, type);
}

```

Wdyt?

Reply:
Something like that makes sense, I really don't like how ipc stuff lacks type checks.